image/mkosi/secure-boot/genkeys.sh

```diff
@@ -0,0 +1,65 @@
+#!/usr/bin/env bash
+
+# This script generates a PKI for secure boot.
+# It is based on the example from https://github.com/systemd/systemd/blob/main/man/loader.conf.xml
+# This is meant to be used for development purposes only.
+# Release images are signed using a different set of keys.
+# Set PKI to an empty folder and PKI_SET to "dev".
+
+SCRIPT_DIR=$( cd -- "$( dirname -- "${BASH_SOURCE[0]}" )" &> /dev/null && pwd )
+TEMPLATES=${SCRIPT_DIR}/templates
+BASE_DIR=$(realpath "${SCRIPT_DIR}/..")
+if [ -z "${PKI}" ]; then
+    PKI=${BASE_DIR}/pki
+fi
+if [ -z "${PKI_SET}" ]; then
+    PKI_SET=dev
+fi
+
+gen_pki () {
+    # Only use for non-production images.
+    # Use real PKI for production images instead.
+    count=`ls -1 ${PKI}/*.{key,crt,cer,esl,auth} 2>/dev/null | wc -l`
+    if [ $count != 0 ]
+    then
+        echo PKI files $(ls -1 $(realpath "--relative-to=$(pwd)" ${PKI})/*.{key,crt,cer,esl,auth}) already exist
+        return
+    fi
+    mkdir -p "${PKI}"
+    pushd "${PKI}"
+
+    uuid=$(systemd-id128 new --uuid)
+    for key in PK KEK db; do
+        openssl req -new -x509 -config "${TEMPLATES}/${PKI_SET}_${key}.conf" -keyout "${key}.key" -out "${key}.crt" -nodes
+        openssl x509 -outform DER -in "${key}.crt" -out "${key}.cer"
+        cert-to-efi-sig-list -g "${uuid}" "${key}.crt" "${key}.esl"
+    done
+
+    for key in MicWinProPCA2011_2011-10-19.crt MicCorUEFCA2011_2011-06-27.crt MicCorKEKCA2011_2011-06-24.crt; do
+        curl -sL "https://www.microsoft.com/pkiops/certs/${key}" --output "${key}"
+        sbsiglist --owner 77fa9abd-0359-4d32-bd60-28f4e78f784b --type x509 --output "${key%crt}esl" "${key}"
+    done
+
+    # Optionally add Microsoft Windows Production CA 2011 (needed to boot into Windows).
+    cat MicWinProPCA2011_2011-10-19.esl >> db.esl
```

Comment:
I don't think we want to boot into Windows for our VMs.

Is this actually required for something or leftovers from the original templates? If required, please adjust comments.

Reply:
Keeping the *default* signing keys used in production environments is wanted. Pinging @thomasten for confirmation.

Reply:
This script is used for dev/debug images only?
What are the pros/cons of keeping the default keys?

Reply:
This script is used by developers to create test keys. The production keys do not exist yet but the secure boot database of the production PKI should be modeled in the exact same way (and also include the same microsoft certs). The only difference should be safer storage of the keys.

> What are the pros/cons of keeping the default keys?

Only con is that you can boot windows using those keys in addition to our own OS. Only pro is to have the same set of keys loaded on all platforms as we can only add our key on Azure (the default keys will always be there).

Reply:
> pro is to have the same set of keys loaded on all platforms

sounds good